metricbeat/module/vsphere/host/host.go

```diff
@@ -111,40 +113,45 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 	}
 
 	for _, hs := range hst {
-		totalCPU := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
-		freeCPU := int64(totalCPU) - int64(hs.Summary.QuickStats.OverallCpuUsage)
-		freeMemory := int64(hs.Summary.Hardware.MemorySize) - (int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024)
-
-		event := common.MapStr{
-			"name": hs.Summary.Config.Name,
-			"cpu": common.MapStr{
-				"used": common.MapStr{
-					"mhz": hs.Summary.QuickStats.OverallCpuUsage,
-				},
-				"total": common.MapStr{
-					"mhz": totalCPU,
-				},
-				"free": common.MapStr{
-					"mhz": freeCPU,
-				},
+		cpu := common.MapStr{
+			"used": common.MapStr{
+				"mhz": hs.Summary.QuickStats.OverallCpuUsage,
 			},
-			"memory": common.MapStr{
-				"used": common.MapStr{
-					"bytes": (int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024),
-				},
-				"total": common.MapStr{
-					"bytes": hs.Summary.Hardware.MemorySize,
-				},
-				"free": common.MapStr{
-					"bytes": freeMemory,
-				},
+		}
+		memory := common.MapStr{
+			"used": common.MapStr{
+				"bytes": int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024,
 			},
 		}
 
+		if hs.Summary.Hardware != nil {
+			totalCPU := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
+			cpu["total"] = common.MapStr{
```

Comment:
What you can use as an alternative here is `cpu.Put('total.mhz', totalCPU)`. It will have the same effect. Using this you could still generate the full event as you had previously on line 118 and then use:

```
event.Put('cpu.total.mhz', totalCPU)
```

The outcome of the code is exactly the same but it would make it shorter and hopefully more readable.

Reply:
Entries have been updated. I quick look will give me some calm brain cycles :smile: